pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -57,15 +57,18 @@
     int numAggregations = aggregationInfos.size();
     int numKeyColumns = numColumns - numAggregations;
 
+    int aggIdx = 0;
     Map<String, Integer> columnIndexMap = new HashMap<>();
+    Map<String, AggregationFunction> aggregationColumnToFunction = new HashMap<>();
     for (int i = 0; i < numColumns; i++) {
-      columnIndexMap.put(dataSchema.getColumnName(i), i);
-    }
-
-    Map<String, AggregationInfo> aggregationColumnToInfo = new HashMap<>();
-    for (AggregationInfo aggregationInfo : aggregationInfos) {
-      String aggregationColumn = AggregationFunctionUtils.getAggregationColumnName(aggregationInfo);
-      aggregationColumnToInfo.put(aggregationColumn, aggregationInfo);
+      String columnName = dataSchema.getColumnName(i).toLowerCase();
+      columnIndexMap.put(columnName, i);
+      if (i >= numKeyColumns) {
+        AggregationFunction aggregationFunction =
```

Comment:
Please add some comments here (key columns are in front of aggregation columns).
Also, `aggIdx === i - numKeyColumns`, so you don't need to maintain another index.

Reply:
There's a comment at the beginning of this method (few lines above this code change).
Fixed the other suggestion